camviewer_ui_impl.py

```diff
@@ -1151,30 +1151,58 @@ def shutdown(self):
 
     def onfileSave(self):
         try:
-            fileName = QFileDialog.getSaveFileName(
+            filename = QFileDialog.getSaveFileName(
                 self,
                 "Save Image...",
-                self.cwd,
+                os.path.expanduser("~"),
                 "Images (*.npy *.jpg *.png *.bmp *.pgm *.tif)",
             )[0]
-            if fileName == "":
+            if filename == "":
                 raise Exception("No File Name Specified")
-            if fileName.lower().endswith(".npy"):
-                np.save(fileName, self.image)
+            if filename.lower().endswith(".npy"):
+                np.save(filename, self.image)
                 QMessageBox.information(
                     self,
                     "File Save Succeeded",
-                    "Image has been saved as a numpy file: %s" % (fileName),
+                    "Image has been saved as a numpy file: %s" % (filename),
                 )
-                print("Saved to a numpy file %s" % (fileName))
-            else:
-                self.ui.display_image.image.save(fileName, format=None, quality=-1)
+                print("Saved to a numpy file %s" % (filename))
+            elif self.ui.display_image.image.save(filename, format=None, quality=-1):
+                # QImage.save returned True, so the save succeeded.
                 QMessageBox.information(
                     self,
                     "File Save Succeeded",
-                    "Image has been saved to an image file: %s" % (fileName),
+                    "Image has been saved to an image file: %s" % (filename),
+                )
+                print("Saved to an image file %s" % (filename))
+            else:
+                # QImage.save returned False, so the save failed.
+                # Check for obvious errors, then retry the save
+                # File already exists? This shouldn't happen due to the qt file dialog.
+                if os.path.exists(filename):
+                    reason = (
+                        f"{filename} already exists! Please pick a different filename"
+                    )
+                # No write permissions?
+                else:
+                    # This is the most robust way to check if we have permissions
+                    # os.access exists but doesn't quite work fully on NFS filesystems as per the Python docs
```

Comment:
I've seen `os.stat` mentioned in conjunction with the `stat` module, if there's a need for an alternate way?  ([stackoverflow](https://stackoverflow.com/questions/1861836/checking-file-permissions-in-linux-with-python))

Reply:
I think `os.stat` also won't fully encompass the NFS permissions but maybe the accuracy drop will be worth the simpler implementation (not accidentally creating a stub file)

Reply:
Resolved via simplifying with `os.access` to reduce compexity

Reply:
Interestingly python suggests the ask-for-forgiveness (try-catch) approach for security reasons we probably don't care about.  

Out of curiosity, do you remember what the NFS limitations were?